services/wfm/src/main/java/org/openkilda/wfm/topology/floodlightrouter/service/RouterService.java

```diff
@@ -0,0 +1,140 @@
+import org.openkilda.messaging.info.InfoMessage;
+import org.openkilda.messaging.info.discovery.NetworkDumpSwitchData;
+import org.openkilda.messaging.info.event.PortInfoData;
+import org.openkilda.messaging.info.event.SwitchInfoData;
+import org.openkilda.model.SwitchId;
+import org.openkilda.wfm.topology.floodlightrouter.Stream;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.UUID;
+
+@Slf4j
+public class RouterService {
+
+    private final FloodlightTracker floodlightTracker;
+
+    public RouterService(FloodlightTracker floodlightTracker) {
+        this.floodlightTracker = floodlightTracker;
+    }
+
+    /**
+     * Process periodic state update.
+     * @param routerMessageSender callback to be used for message sending
+     */
+    public void doPeriodicProcessing(MessageSender routerMessageSender) {
+        for (String region : floodlightTracker.getRegionsForAliveRequest()) {
+            AliveRequest request = new AliveRequest();
+            CommandMessage message = new CommandMessage(request, System.currentTimeMillis(), UUID.randomUUID()
+                    .toString());
+            routerMessageSender.send(message, Stream.formatWithRegion(Stream.SPEAKER_DISCO, region));
+
+        }
+        floodlightTracker.checkTimeouts();
+        floodlightTracker.handleUnmanagedSwitches(routerMessageSender);
+    }
+
+    /**
+     * Process response from speaker disco.
+     * @param routerMessageSender callback to be used for message sending
+     * @param message message to be handled and resend
+     */
+    public void processSpeakerDiscoResponse(MessageSender routerMessageSender,
+                                            Message message) {
+        if (message instanceof InfoMessage) {
+            InfoMessage infoMessage = (InfoMessage) message;
+            InfoData infoData = infoMessage.getData();
+            SwitchId switchId = null;
+            String region = ((InfoMessage) message).getRegion();
+            handleResponseFromSpeaker(routerMessageSender, region, message.getTimestamp());
+            if (infoData instanceof AliveResponse) {
+                AliveResponse aliveResponse = (AliveResponse) infoData;
+                if (aliveResponse.getFailedMessages() > 0) {
+                    sendNetworkRequest(routerMessageSender, region);
+                }
+                return;
+            } else if (infoData instanceof NetworkDumpSwitchData) {
+                switchId = ((NetworkDumpSwitchData) infoData).getSwitchRecord().getDatapath();
+            } else if (infoData instanceof SwitchInfoData) {
+                switchId = ((SwitchInfoData) infoData).getSwitchId();
+            } else if (infoData instanceof PortInfoData) {
+                switchId = ((PortInfoData) infoData).getSwitchId();
+            }
+
+            // NOTE(tdurakov): need to notify of a mapping update
+            if (switchId != null && region != null && floodlightTracker.updateSwitchRegion(switchId, region)) {
```

Comment:
1. Is it a possible scenario that we can get a response from speaker without region?
2. Why we always update mapping between a region and a switch? I believe we don't have to do it so often, only do it if it was changed.

Reply:
1 - no, but this validation should be here
2 - it's important actually. Regions would be set up dynamically, so this regular update allows not to redeploy topology once switches change region